### ISSUE TITLE
[Feature request] Let subplot active "next" panel

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#567]
``subplot`` is very convenient for plotting multiple panels in one figure. Usually, I want to plot similar maps in different panels using a **for loop**. For current implementation of subplot, 
I have to define variables `row` and `col` and increase `row` and/or `col` explicitly to active a panel. 

It would be more flexible if `subplot set` and `-c` option can active the **next panel** in a smart way.

This is an example showing a possible implementation of this feature and how this feature can be helpful. In this example, I want to plot four basemaps with different colors filled. 
```
gmt begin subplot png
gmt subplot begin 2x2 -Fs5c/3c -A'(a)'+JTL+o0.2c/0.4c -M0.5c/0.2c
for color in blue red lightblue lightred
do
    gmt basemap -R0/5/0/5 -Ba -BWSen+g$color -c0,0
    echo 3 3 | gmt plot -Sc1c 
done
gmt subplot end
gmt end
```
Here, invalid subplot panel `0,0` is used to active the next panel. In the four iterations, `-c0,0` automatically activates four panels in the order of 1,1 => 1,2 => 2,1 => 2,2 (row first) or 1,1 => 2,1 => 1,2 => 2,2 (column first).

**Comment by PaulWessel** [comment]
How about just **-c** with no arguments?

**Comment by seisman** [comment]
`-c` without argument looks fine. But it would be better if users can choose row-first or column-first order. And what about `subplot set`? We should the have same syntax of `subplot set` and `-c` for activating the "next" panel.

**Comment by PaulWessel** [comment]
The **-A+v** modifier numbers down columns so presumably if you are looping and want to go down columns then you want to number them that way too, i.e., increment some counter to get 1, 2, 3 or a, b, c.  So let the **-A** setting determine what the "next" panel is [Default is row-wise]. And gmt subplot set without a row, col argument should behave the same way.  Finally, if there is no previous panel set then we default to the first panel.

**Comment by seisman** [comment]
Closed in #569.
